src/Marten/Linq/Parsing/SimpleBinaryComparisonExpressionParser.cs

```diff
@@ -34,29 +34,30 @@ public bool Matches(BinaryExpression expression)
 
         public IWhereFragment Parse(IQueryableDocument mapping, ISerializer serializer, BinaryExpression expression)
         {
-            var isValueExpressionOnRight = expression.Right.IsValueExpression();
+            var areBothMemberExpresions = !expression.Left.IsValueExpression() && !expression.Right.IsValueExpression();
+            var isValueExpressionOnRight = areBothMemberExpresions || expression.Right.IsValueExpression();
             var jsonLocatorExpression = isValueExpressionOnRight ? expression.Left : expression.Right;
             var valueExpression = isValueExpressionOnRight ? expression.Right : expression.Left;
 
             var members = FindMembers.Determine(jsonLocatorExpression);
 
-            var field = mapping.FieldFor(members);
-
-	        object value;
-
-	        if (valueExpression is MemberExpression memberAccess)
-	        {
-		        var membersOther = FindMembers.Determine(memberAccess);
-		        var fieldOther = mapping.FieldFor(membersOther);
-		        value = fieldOther.SqlLocator;
-	        }
-	        else
-	        {
-		        memberAccess = null;
-				value = field.GetValue(valueExpression);
-	        }
-
-	        var jsonLocator = field.SqlLocator;
+            var field = mapping.FieldFor(members);

```

Comment:
There are no changes in this code path, not sure why it is showing a diff

Reply:
@mysticmind I have autoformatting turned on in my visual studio. It seems that file had previously mixed tabs and spaces. Now it fixed it to have spaces.